core/lib/state/src/postgres/mod.rs

```diff
@@ -393,6 +393,11 @@ impl<'a> PostgresStorage<'a> {
     fn values_cache(&self) -> Option<&ValuesCache> {
         Some(&self.caches.as_ref()?.values.as_ref()?.cache)
     }
+
+    /// Get the connection to the database.
+    pub fn connection(&mut self) -> &mut StorageProcessor<'a> {
+        &mut self.connection
+    }
```

Comment:
PostgresStorage assumes a unique ownership of the connection. I don't think we want to expose the underlying connection (generally, exposing something internal while there is a way to obtain the same resource otherwise is usually not a good idea).